source/Halibut.Tests/TraceLogFileLogger.cs

```diff
@@ -1,25 +1,82 @@
 ﻿using System;
+using System.Collections.Generic;
 using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Threading;
 
 namespace Halibut.Tests
 {
     public class TraceLogFileLogger : IDisposable
     {
-        string tempFilePath = Path.GetTempFileName();
+        readonly AsyncQueue<string> queue = new AsyncQueue<string>();
+        readonly string tempFilePath = Path.GetTempFileName();
         string testHash;
 
+        readonly TaskCompletionSource<bool> noMoreDelaysToWritingLogs;
+        readonly Task noMoreDelaysTask;
+        readonly Task writeDataToDiskTask;
+
+        public TraceLogFileLogger()
+        {
+            noMoreDelaysToWritingLogs = new TaskCompletionSource<bool>();
+            noMoreDelaysTask = noMoreDelaysToWritingLogs.Task;
+            writeDataToDiskTask = WriteDataToFile();
+        }
+
         public void SetTestHash(string testHash)
         {
             this.testHash = testHash;
         }
         
         public void WriteLine(string logMessage)
         {
-            File.AppendAllLines(tempFilePath, new[] { logMessage });
+            queue.Enqueue(logMessage);
+        } 
+        
+        async Task WriteDataToFile()
```

Comment:
This task is never terminated, my guess it wont be an issue since we don't have enough tests for this to become an issue.